From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Dreeam <61569423+Dreeam-qwq@users.noreply.github.com>
Date: Fri, 11 Jul 2025 06:10:41 +0800
Subject: [PATCH] Fix infinite loop in dismount loc check

Removed since Leaf 1.21.7, fixed in Paper (mache) side,
Very fun VineFlower decompile issue.

Invert the condition to prevent infinite loop

diff --git a/net/minecraft/world/entity/animal/horse/AbstractHorse.java b/net/minecraft/world/entity/animal/horse/AbstractHorse.java
index e78ef459d6eecb7f69882ed95cd9f9967f53989f..3bdb6afe3acceb37173b1aab3d2d22a96b53d046 100644
--- a/net/minecraft/world/entity/animal/horse/AbstractHorse.java
+++ b/net/minecraft/world/entity/animal/horse/AbstractHorse.java
@@ -1056,7 +1056,7 @@ public abstract class AbstractHorse extends Animal implements HasCustomInventory
                 }
 
                 mutableBlockPos.move(Direction.UP);
-            } while (!(mutableBlockPos.getY() < d3));
+            } while (mutableBlockPos.getY() < d3); // Leaf - Fix infinite loop in dismount loc check
         }
 
         return null;
